src/approaches/grammar_search_invention_approach.py

```diff
@@ -469,6 +469,26 @@ def enumerate(self) -> Iterator[Tuple[Predicate, float]]:
 class _PrunedGrammar(_DataBasedPredicateGrammar):
     """A grammar that prunes redundant predicates."""
     base_grammar: _PredicateGrammar
+    _state_sequences: List[List[State]] = field(init=False,
+                                                default_factory=list)
+
+    def __post_init__(self) -> None:
+        if CFG.segmenter == "option_changes":
+            # If the segmenter is based on changes in options, then
+            # because it doesn't depend on atoms, we can be very
+            # efficient during pruning by pre-computing the segments.
+            # Then, we only need to care about the initial and final
+            # states in each segment, which we store into
+            # self._state_sequence.
+            atom_dataset = utils.create_ground_atom_dataset(
```

Comment:
why are we doing this? is atom_dataset used?

Reply:
isn't it used in the next line? we just need an atom_dataset to be able to call segment_trajectory

Reply:
oh, you think we should just pass in an empty thing there, i see, okay!

Reply:
ah okay got it